app/models.py

```diff
@@ -20,6 +21,23 @@ class DatasetModel(db.DynamicDocument):
     deleted = db.BooleanField(default=False)
     deleted_date = db.DateTimeField()
 
+    @classmethod
+    def create_from_json(cls, json_file):
+        with open(json_file) as file:
+            datasets = json.load(file)
+            for dataset_json in datasets:
+                name = dataset_json.get('name')
+                if name:
+                    try:
+                        dataset = DatasetModel(
+                            name=name, 
+                            categories=dataset_json.get('categories', []))
```

Comment:
Categories are stored as a list of ids (not names). You'll have to do an instance check, get the ids of named categories, and create any missing categories like [here](https://github.com/jsbroks/coco-annotator/blob/master/app/api/datasets.py#L66).